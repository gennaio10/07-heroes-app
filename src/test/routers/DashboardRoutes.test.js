import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas de <DashboardRoutes />)", () => {
  const defaultName = 'Pororo'

  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: true, name: defaultName },
  };

  test("debe de mostrar correctamente", () => {
    const wrapper = mount(
      <MemoryRouter>
        <AuthContext.Provider value={contextValue}>
          <DashboardRoutes />
        </AuthContext.Provider>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text().trim()).toBe(defaultName);
  });
});
