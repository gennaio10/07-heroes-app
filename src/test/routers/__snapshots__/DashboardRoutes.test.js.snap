// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <DashboardRoutes />) debe de mostrar correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Searh
        </a>
      </div>
    </div>
    <div
      className="navbar-text"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-into"
        >
          Pororo
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
        MarvelScreen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-4 animate__animated animate__fadeIn"
      >
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Spider Man"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                Peter Parker
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Amazing Fantasy #15
                </small>
              </p>
              <a
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Captain America"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                Steve Rogers
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Captain America Comics #1
                </small>
              </p>
              <a
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Iron Man"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                Tony Stark
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Tales of Suspense #39
                </small>
              </p>
              <a
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Thor"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                Thor Odinson
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Journey into Myster #83
                </small>
              </p>
              <a
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Hulk"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                Bruce Banner
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #1
                </small>
              </p>
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Wolverine"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                James Howlett
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #180
                </small>
              </p>
              <a
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Daredevil"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                Matthew Michael Murdock
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Daredevil #1
                </small>
              </p>
              <a
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Hawkeye"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                Clinton Francis Barton
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Tales of Suspense #57
                </small>
              </p>
              <a
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Cyclops"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                Scott Summers
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  X-Men #1
                </small>
              </p>
              <a
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        <div
          className="card mb-3"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="Silver Surfer"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                Norrin Radd
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Fantastic Four #48
                </small>
              </p>
              <a
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
